src/pages/Register.js

```diff
@@ -1,93 +1,133 @@
 import React from "react";
 import { Navigate, Link } from "react-router-dom";
+import { useForm } from "react-hook-form";
+import { useMutation } from "react-query";
+import {
+  TextField,
+  Button,
+  Typography,
+  CircularProgress,
+  Alert,
+} from "@mui/material";
 import { useAuth } from "../context/authContext/index";
-import { useMutation } from "react-query"; // Import QueryClient and QueryClientProvider
 import { authCreateUser } from "../firebase/auth";
 
 const Register = () => {
-  const { userLoggedIn } = useAuth();
+  const { isLoggedIn } = useAuth();
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm();
+  const password = React.useRef({});
+  password.current = watch("password", "");
```

Comment:
You can listen the value directly with `watch` and providing the registered name to the method. 
`const password = watch("password");`